rust/integration-tests/src/sdk_tests/sdk_test_helpers.rs

```diff
@@ -0,0 +1,181 @@
+    }
+
+    /// Helper function to set up and run the mock GRPC server.
+    async fn setup_mock_grpc(transactions: Vec<TransactionsResponse>, chain_id: u64) {
+        let mock_grpc_server = MockGrpcServer {
+            transactions,
+            chain_id,
+        };
+
+        // Start the Mock GRPC server
+        tokio::spawn(async move {
+            println!("Starting Mock GRPC server");
+            mock_grpc_server.run().await;
+        });
+    }
+
+    /// Helper function to initialize the Events Processor.
+    async fn init_events_processor(db_url: String) -> EventsProcessor {
+        let events_processor_config = EventsProcessorConfig {
+            channel_size: 10,
+            per_table_chunk_sizes: AHashMap::new(),
+        };
+
+        let processor_config = ProcessorConfig::EventsProcessor(events_processor_config);
+
+        let transaction_stream_config = TransactionStreamConfig {
+            indexer_grpc_data_service_address: Url::parse("http://localhost:51254").expect("Could not parse database url"),
+            starting_version: Some(1), // starting version for the stream
+            request_ending_version: Some(1),
+            auth_token: "".to_string(),
+            request_name_header: "".to_string(),
+            indexer_grpc_http2_ping_interval_secs: 30,
+            indexer_grpc_http2_ping_timeout_secs: 10,
+            indexer_grpc_reconnection_timeout_secs: 10,
+            indexer_grpc_response_item_timeout_secs: 60,
+        };
+
+        let postgres_config = PostgresConfig {
+            connection_string: db_url.clone(),
+            db_pool_size: 100,
+        };
+
+        let db_config = DbConfig::PostgresConfig(postgres_config);
+
+        let indexer_processor_config = IndexerProcessorConfig {
+            processor_config,
+            transaction_stream_config,
+            db_config,
+        };
+
+        EventsProcessor::new(indexer_processor_config).await.expect("Failed to create EventsProcessor")
+    }
+
+    #[tokio::test]
+    async fn test_run() {
+        // Setup Postgres
+        let (db_url, _postgres_container) = setup_postgres().await;
+
+        // Setup Mock GRPC server
+        let transaction = Transaction {
+            version: 1,
+            ..Transaction::default()
+        };
+        let transactions = vec![TransactionsResponse {
+            transactions: vec![transaction],
+            ..TransactionsResponse::default()
+        }];
+        setup_mock_grpc(transactions, 1).await;
+
+        // Initialize and run the EventsProcessor
+        let events_processor = init_events_processor(db_url.clone()).await;
+        events_processor.run_processor().await.expect("Failed to run processor");
+
+        // Connect to Postgres and verify the events
+        let mut conn = PgConnection::establish(&db_url)
+            .with_context(|| format!("Error connecting to {}", db_url)).expect("Failed to connect to database");
+
+        let events_result = events
```

Comment:
somehow this doesn't return any events. @larry-aptos do you have any insights how I can debug this?

Reply:
It might be that the transaction didn't process yet. Since with SDK all the work is split between separate tasks, it takes some time to process. We can add `tokio::time::sleep(Duration::from_millis(250)).await` to wait to finish processing before checking the DB.